src/main/java/net/querz/mcaselector/version/NbtHelper.java

```diff
@@ -253,4 +258,106 @@ public static int findHighestSection(ListTag sections, int lowest) {
+					result.add(dest);
+				}
+			}
+		}
+
+		for (Tag sourceElement : source) {
+			int y = ySupplier.apply(sourceElement);
+			for (Range range : ranges) {
+				if (range.contains(y - yOffset)) {
+					result.add(sourceElement);
+					break;
+				}
+			}
+		}
+
+		return result;
+	}
+
+	public static void mergeListTagLists(CompoundTag source, CompoundTag destination, List<Range> ranges, int yOffset, String name) {
+		ListTag sourceList = tagFromLevelFromRoot(source, name);
+		ListTag destinationList = tagFromLevelFromRoot(destination, name, sourceList);
+
+		if (sourceList == null || destinationList == null || sourceList.size() != destinationList.size()) {
+			return;
+		}
+
+		for (Range range : ranges) {
+			int m = Math.min(range.getTo() + yOffset, sourceList.size() - 1);
+			for (int i = Math.max(range.getFrom() + yOffset, 0); i <= m; i++) {
+				destinationList.set(i, sourceList.get(i));
+			}
+		}
+
+		initLevel(destination).put(name, destinationList);
+	}
+
+	public static void mergeCompoundTagListsFromLevel(CompoundTag source, CompoundTag destination, List<Range> ranges, int yOffset, String name, Function<Tag, Integer> ySupplier) {
+		ListTag sourceElements = tagFromLevelFromRoot(source, name, new ListTag());
+		ListTag destinationElements = tagFromLevelFromRoot(destination, name, new ListTag());
+
+		initLevel(destination).put(name, mergeLists(sourceElements, destinationElements, ranges, ySupplier, yOffset));
+	}
+
+	public static void mergeCompoundTagLists(CompoundTag source, CompoundTag destination, List<Range> ranges, int yOffset, String name, Function<Tag, Integer> ySupplier) {
+		ListTag sourceElements = tagFromCompound(source, name, new ListTag());
+		ListTag destinationElements = tagFromCompound(destination, name, new ListTag());
+
+		destination.put(name, mergeLists(sourceElements, destinationElements, ranges, ySupplier, yOffset));
+	}
+
+	// merge based on compound tag keys, assuming compound tag keys are ints
+	public static void mergeCompoundTags(CompoundTag source, CompoundTag destination, List<Range> ranges, int yOffset, String name) {
+		CompoundTag sourceElements = tagFromCompound(source, name, new CompoundTag());
+		CompoundTag destinationElements = tagFromCompound(destination, name, new CompoundTag());
+
+		for (Map.Entry<String, Tag> sourceElement : sourceElements) {
+			if (sourceElement.getKey().matches("^-?[0-9]{1,2}$")) {
+				int y = Integer.parseInt(sourceElement.getKey());
+				for (Range range : ranges) {
+					if (range.contains(y - yOffset)) {
+						destinationElements.put(sourceElement.getKey(), sourceElement.getValue());
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	public static CompoundTag initLevel(CompoundTag c) {
+		CompoundTag level = levelFromRoot(c);
+		if (level == null) {
+			c.put("Level", level = new CompoundTag());
+		}
+		return level;
+	}
+
+	// XXX what's going on with the colliding fixEntityUUID* methods? What do they actually do? How can they be named better?
```

Comment:
`fixEntityUUIDsMerger` handles entities on the root level, while `fixEntityUUIDMerger` is used to recursively change entity UUIDs for any passengers.

I guess they could be renamed to something like `recalculateEntityUUID` and `recalculateEntityUUIDs` to better describe what they do.